notebooks/concepts/glue_units_poc.ipynb

```diff
@@ -0,0 +1,177 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "10c31fd2-4046-4d61-8ae0-30cb8344d41f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from astropy.wcs import WCS\n",
+    "import glue_jupyter as gj\n",
+    "from glue.core import Data\n",
+    "from glue.plugins.wcs_autolinking.wcs_autolinking import WCSLink\n",
+    "from specutils import Spectrum1D\n",
+    "from glue.core.link_helpers import LinkSame"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8d80be95-e29f-4f3f-8bee-6a625c291473",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wcs1 = WCS(naxis=1)\n",
+    "wcs1.wcs.ctype = ['WAVE']\n",
+    "wcs1.wcs.crval = [10]\n",
+    "wcs1.wcs.cdelt = [3]\n",
+    "wcs1.wcs.crpix = [1]\n",
+    "wcs1.wcs.cunit = ['cm']\n",
+    "\n",
+    "d1 = Data(temp1=np.random.random(10))\n",
+    "d1.get_component('temp1').units = 'deg_C'\n",
+    "d1.coords = wcs1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ab5eb050-d904-48de-9fc2-d045e6252a4d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wcs1 = WCS(naxis=1)\n",
+    "wcs1.wcs.ctype = ['WAVE']\n",
+    "wcs1.wcs.crval = [10]\n",
+    "wcs1.wcs.cdelt = [3]\n",
+    "wcs1.wcs.crpix = [1]\n",
+    "wcs1.wcs.cunit = ['cm']\n",
+    "\n",
+    "d2 = Data(temp2=np.random.random(10) + 273)\n",
+    "d2.get_component('temp2').units = 'K'\n",
+    "d2.coords = wcs1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "458dea14-6eb0-4aea-a6ad-c374deca1a1f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from astropy import units as u\n",
+    "from glue.core.units import unit_converter\n",
+    "\n",
+    "@unit_converter('custom-astropy')\n",
+    "class UnitConverterWithTemperature:\n",
+    "\n",
+    "    def equivalent_units(self, data, cid, units):\n",
+    "        equivalencies = u.temperature() if 'temp' in cid.label.lower() else None\n",
```

Comment:
how will this work in practice (we don't want all equivalencies for all axes)?  Is the lack of the spectral equivalency here the reason Hz is not converting as expected?

Reply:
Both good questions. I will look into the second one, no good answer for the first one.

Reply:
Maybe a lookup dictionary of some sort to map conv type to equivalencies, or a bunch of if-else logic blocks?

FWIW, if-else is applied at https://github.com/spacetelescope/synphot_refactor/blob/2a382d7bdf29cc4a1e6b69e59d5c1d0f82dabffc/synphot/units.py#L242

Reply:
The reason this doesn't work currently is because you should indeed be using the spectral equivalency for some components - here's a simple example that works for this particular example:

```
from astropy import units as u
from glue.core.units import unit_converter

@unit_converter('custom-astropy')
class UnitConverterWithSpectral:

    def equivalent_units(self, data, cid, units):
        equivalencies = u.spectral() if cid.label.lower() in ['freq', 'wave'] else None
        return map(str, u.Unit(units).find_equivalent_units(include_prefix_units=True, equivalencies=equivalencies))

    def to_unit(self, data, cid, values, original_units, target_units):
        equivalencies = u.spectral() if cid.label.lower() in ['freq', 'wave'] else None
        return (values * u.Unit(original_units)).to_value(target_units, equivalencies=equivalencies)
```